### PR TITLE
Feature/short paths export sources

### PR DESCRIPTION
**Pull request by memsharded**
Implements fully #1299

### CODE REVIEW

File: conans/client/client_cache.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def conan_manifests(self, conan_reference):
         digest_path = self.digestfile_conanfile(conan_reference)
         if not os.path.exists(digest_path):
             return None, None
-        export_sources_path = self.export_sources(conan_reference)
+        export_sources_path = self.export_sources(conan_reference, short_paths=None)
```
Comment (lasote) [inline conans/client/client_cache.py]: Why None? I don't get why is not needed here the short paths check

Comment (memsharded) [inline conans/client/client_cache.py]: ``short_paths=None`` means use short paths if the link already exists. It is not that it is not needed it is that the folder should exist, and the link should be there if short_paths

Comment (lasote) [inline conans/client/client_cache.py]: Ah. Got it. Is it commented somewhere in code?

Comment (memsharded) [inline conans/client/client_cache.py]: It was the previous behavior of ``short_paths``,  it is documented in the ``_shortener`` function,  the one doing the work.

**Pull request merged**